ferrocene/doc/requirements/src/self-test.rst

```diff
@@ -0,0 +1,209 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+`ferrocene-self-test` tool
+==========================
+
+General checks
+--------------
+
+Checks for executables
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+   **executable** refers to the following:
+
+   - ``rustc``
+   - ``rustdoc``
+
+The executable should exist in the ``bin`` directory, relative to the root of the installation.
+
+The metadata of the executable should be fetchable.
+
+The executable should be a regular file.
+
+The executable should have read and execute permissions set for all users.
+
+When executed with ``-vV`` command-line options, the following apply:
+
+- The executable should exit successfully.
+
+- The output of the executable should be UTF-8 text.
+
+- The output of the executable should have at least 3 values: host, commit-hash, and release.
+
+- The value of ``host`` from the output of the executable should match the target-triple
+  of the executing environment.
+
+- The value of ``commit-hash`` from the output of the executable should match the 
+  commit hash of the Ferrocene release corresponding to the installed Ferrocene toolchain.
+
+- The value of ``release`` from the output of the executable should match the
+  version of the Ferrocene release corresponding to the installed Ferrocene toolchain.
+
+Checks for targets
```

Comment:
We don't mention anywhere that only targets that are actually installed are checked.

Reply:
f16e9625b75